contrib/lib/src/databases.rs

```diff
@@ -0,0 +1,209 @@
+
+    Ok(db_config
+        .iter()
+        .filter_map(|config| config.as_table())
+        .filter_map(|table| table.get("name"))
+        .filter_map(|val| val.as_str()))
+}
+
+/// This trait can be implemented on Connection types that can be stored in a connection pool.
+/// Implementing this trait on a type allows Rocket to generate the necessary code to manage the
+/// pool internally and inject it into request handlers.
+pub trait Poolable: Send + Sized + 'static {
+    type Manager: ManageConnection<Connection=Self>;
+    type Error;
+
+    fn pool(config: DatabaseConfig) -> Result<r2d2::Pool<Self::Manager>, Self::Error>;
+}
+
+#[cfg(feature = "diesel_sqlite_pool")]
+impl Poolable for diesel::SqliteConnection {
+    type Manager = diesel::r2d2::ConnectionManager<diesel::SqliteConnection>;
+    type Error = r2d2::Error;
+
+    fn pool(config: DatabaseConfig) -> Result<r2d2::Pool<Self::Manager>, Self::Error> {
+        let manager = diesel::r2d2::ConnectionManager::new(config.url);
+        r2d2::Pool::builder().max_size(config.pool_size as u32).build(manager)
+    }
+}
+
+#[cfg(feature = "diesel_pg_pool")]
+impl Poolable for diesel::PgConnection {
+    type Manager = diesel::r2d2::ConnectionManager<diesel::PgConnection>;
+    type Error = r2d2::Error;
+
+    fn pool(config: DatabaseConfig) -> Result<r2d2::Pool<Self::Manager>, Self::Error> {
+        let manager = diesel::r2d2::ConnectionManager::new(config.url);
+        r2d2::Pool::builder().max_size(config.pool_size as u32).build(manager)
+    }
+}
+
+#[cfg(feature = "diesel_mysql_pool")]
+impl Poolable for diesel::MysqlConnection {
+    type Manager = diesel::r2d2::ConnectionManager<diesel::MysqlConnection>;
+    type Error = r2d2::Error;
+
+    fn pool(config: DatabaseConfig) -> Result<r2d2::Pool<Self::Manager>, Self::Error> {
+        let manager = diesel::r2d2::ConnectionManager::new(config.url);
+        r2d2::Pool::builder().max_size(config.pool_size as u32).build(manager)
+    }
+}
+
+// TODO: Come up with a way to handle TLS
+#[cfg(feature = "postgres_pool")]
+impl Poolable for postgres::Connection {
+    type Manager = r2d2_postgres::PostgresConnectionManager;
+    type Error = DbError<postgres::Error>;
+
+    fn pool(config: DatabaseConfig) -> Result<r2d2::Pool<Self::Manager>, Self::Error> {
+        let manager = r2d2_postgres::PostgresConnectionManager::new(config.url, r2d2_postgres::TlsMode::None)
+            .map_err(DbError::Custom)?;
+
+        r2d2::Pool::builder().max_size(config.pool_size as u32).build(manager)
+            .map_err(DbError::PoolError)
+    }
+}
+
+#[cfg(feature = "redis_pool")]
+impl Poolable for redis::Connection {
+    type Manager = r2d2_redis::RedisConnectionManager;
+    type Error = DbError<redis::RedisError>;
+
+    fn pool(config: DatabaseConfig) -> Result<r2d2::Pool<Self::Manager>, Self::Error> {
+        r2d2_redis::RedisConnectionManager::new(config.url).map_err(DbError::Custom)?;
```

Comment:
CI failed, I believe because this line needs a `let manager =`

Reply:
Yeah... my testing is not exhaustive because of these `cfg` flags... *sigh* I need to be more careful...

Reply:
Why not run the test script yourself? `./scripts/test.sh && ./scripts/test.sh --release`. That's all the CI does.